ksql-serde/src/test/java/io/confluent/ksql/serde/avro/KsqlGenericRowAvroDeserializerTest.java

```diff
@@ -201,4 +213,342 @@ public void shouldDeserializeWithMissingFields() {
 
     return kafkaAvroSerializer.serialize(topicName, avroRecord);
   }
+
+  private void shouldDeserializeTypeCorrectly(final org.apache.avro.Schema avroSchema,
+                                              final Object avroValue,
+                                              final Schema ksqlSchema) {
+    shouldDeserializeTypeCorrectly(avroSchema, avroValue, ksqlSchema, avroValue);
```

Comment:
Why do we use `avroValue` as `ksqlValue`?

Reply:
For many cases the expected value is the same as the value in the avro record.